mt76: replace strlcpy with strscpy

namiltd opened the pull request:
Change deprecated function strlcpy to strscpy for compatibility with kernel 6.12.

strlcpy() reads the entire source buffer first. This read may exceed the destination size limit.
This is both inefficient and can lead to linear read overflows if a source string is not NUL-terminated [1].
In an effort to remove strlcpy() completely [2], replace strlcpy() here with strscpy().

Direct replacement is safe here since DEV_ASSIGN is only used by TRACE macros and the return values are ignored.

[1] https://www.kernel.org/doc/html/latest/process/deprecated.html#strlcpy
[2] https://github.com/KSPP/linux/issues/89

castiel652 commented:
This is actually fixed already in upstream.
Maybe we can ask Felix to sync it?
https://web.git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/commit/?id=d6b484b5cb2a7d509b36a220911509ddd8b777c4

namiltd committed:
mt76: replace strlcpy with strscpy

Change deprecated function strlcpy to strscpy
for compatibility with kernel 6.12.

strlcpy() reads the entire source buffer first.
This read may exceed the destination size limit.
This is both inefficient and can lead to linear read
overflows if a source string is not NUL-terminated [1].
In an effort to remove strlcpy() completely [2], replace
strlcpy() here with strscpy().

Direct replacement is safe here since DEV_ASSIGN is only used by
TRACE macros and the return values are ignored.

[1] https://www.kernel.org/doc/html/latest/process/deprecated.html#strlcpy
[2] https://github.com/KSPP/linux/issues/89

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Link: https://github.com/openwrt/openwrt/pull/18927
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

rmandrad commented:
> This is actually fixed already in upstream. Maybe we can ask Felix to sync it? https://web.git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/commit/?id=d6b484b5cb2a7d509b36a220911509ddd8b777c4

yes mt76 needs a refresh from upstream

libiunc commented:
Tested on my Cudy TR3000 v1 and Netcore N60 Pro, with correct expected behavior

namiltd commented:
@libiunc Replacing strlcpy with strscpy is simpler.

libiunc commented:
Tested on my Cudy TR3000 v1, Netcore N60 Pro, Radxa ROCK 3B, Bananapi BPi-R64, StarFive VisionFive2 v1.3b and has correct expected behavior(mt7615,mt7915,mt76x2,mt7921,mt7922)
